### DISCUSSION TITLE
Evaluating custom trained model

### DISCUSSION THREAD

**Discussion by sneakybatman** [mindee/doctr#1827] [Q&A] [answered]
Hi, I am trained a model on my custom dataset using "master" architecture. The results during training are promising on the validation dataset. I have written a script to run prediction on my test set with the new custom trained model. I was curious if there is some script already existing where I could pass the trained model path and test set path which then does the complete evaluation on the test set?

**Comment by felixdittrich92** [comment, marked as answer]
Hi @sneakybatman 👋 ,

We have an script for E2E evaluation here: https://github.com/mindee/doctr/blob/main/scripts/evaluate.py
And only for recognition here:
https://github.com/mindee/doctr/blob/main/references/recognition/evaluate_pytorch.py

Currently designed to work with our ready to use built-in datasets but you could easily modify both scripts to work with your custom model / dataset 

Best,
Felix :)

**Comment by sneakybatman** [comment]
Thanks for the direction @felixdittrich92 . I adapted it to use my dataset and it works. Happy Holidays :)

**Reply by felixdittrich92** [reply]
Wish you too thanks 😊
